modules/lvm2/data/org.freedesktop.UDisks2.lvm2.xml

```diff
@@ -302,7 +302,8 @@
          type on the given physical volumes.  It is okay to leave
          "pvs" empty; LVM2 will then choose suitable ones on its own.
 
-         No additional options are currently defined.
+         The "stripes" (u) and "mirrors" (u) options work like the
+         "stripes" and "mirrors" options of "lvcreate".
```

Comment:
Yes, I was pondering this as well.

So, if there is time to get this into the same release as CreatePlainVolumeWithLayouts, we can still change the default behavior of that method, and try to improve it.

It seems to be important to never let lvcreate create a raid volume that has less subvolumes than the physical volumes given to lvcreate, because of a bug in the lvcreate allocation algoritm (https://bugzilla.redhat.com/show_bug.cgi?id=2181573). That was the immediate motivation to finally do something about the `--stripes` and `--mirrors` options.

And as you say, doing this in the most straightforward way, as in the initial version of this PR, exposes a lot of LVM2 details in the API.  For example, to get 4 subvolumes for a raid5, one has to pass `--stripes 3`, but for a four subvolume raid0 it is `--stripes 4`. 

However, we can change CreatePlainWithLayout to work around the allocator bug by default: We can make it create a logical volume with the same number of subvolumes as the number of PVs.  Cockpit would just use that default initially.

We could add an option "subvolumes" that can tweak this, and strongly advise against it unless the user knows what might happen.

This would be a lot more logic in UDisks2, and we would deviate from the default behavior of lvcreate. But I think that's ok.  The UDisks2 API has always tried to paper over the tricky bits.

What do you think?

Reply:
(I code this up as a separate PR, for easier comparison.)